plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -1005,6 +1011,34 @@ def verify_size(self, location):
             with open(location) as fp:
                 verification.verify_size(fp, self.size)
 
+    def set_repodata(self, metadata_type, xml_snippet):
```

Comment:
`repodata` was a good candidate for property with getter/setter but the solution required some sort of cache(?) because repodata is dictionary and I guess we do not want to do compression/decompression more times than needed and for all of the repodata. So I ended up with just simple methods but it is not possible to set/get all the repo data at once, just one by one. 
I am open to any better suggestions/solutions here.